pkg/haproxy/types.go

```diff
@@ -56,6 +56,7 @@ type HTTPService struct {
 	NodePort      int32
 	OffloadSSL    bool
 	FrontendRules []string
+	TLSAuth       *api.TLSAuth
```

Comment:
Don't use the api object directly. Read the secret and validate that ca.crt actually exists. If not, return error. Then set the CAFile to `secret-ca.pem`. Then we can also add CRL file later.